db/sql/cassandra/connection.go

```diff
@@ -0,0 +1,122 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cassandra
+
+import (
+	"strings"
+	"time"
+
+	"github.com/gocql/gocql"
+)
+
+// Config Configuration for Cassandra clients loaded from a configuration file
+type Config struct {
+
+	// A list of host addresses of cluster nodes.
+	Hosts []string `json:"hosts"`
+
+	// port for Cassandra (default: 9042)
+	Port int `json:"port"`
+
+	// connection timeout (default: 600ms)
+	Timeout time.Duration `json:"timeout"`
+
+	// initial connection timeout, used during initial dial to server (default: 600ms)
+	ConnectTimeout time.Duration `json:"connect_timeout"`
+
+	// If not zero, gocql attempt to reconnect known DOWN nodes in every ReconnectSleep.
+	ReconnectInterval time.Duration `json:"reconnect_interval"`
+
+	// ProtoVersion sets the version of the native protocol to use, this will
+	// enable features in the driver for specific protocol versions, generally this
+	// should be set to a known version (2,3,4) for the cluster being connected to.
+	//
+	// If it is 0 or unset (the default) then the driver will attempt to discover the
+	// highest supported protocol for the cluster. In clusters with nodes of different
+	// versions the protocol selected is not defined (ie, it can be any of the supported in the cluster)
+	ProtoVersion int `json:"proto_version"`
+}
+
+// ClientConfig wrapping gocql ClusterConfig
+type ClientConfig struct {
+	*gocql.ClusterConfig
+}
+
+const defaultTimeout = 600 * time.Millisecond
+const defaultConnectTimeout = 600 * time.Millisecond
+const defaultReocnnectInterval = 60 * time.Second
+const defaultProtoVersion = 4
+
+// ConfigToClientConfig transforms the yaml configuration into ClientConfig.
+func ConfigToClientConfig(ymlConfig *Config) (*ClientConfig, error) {
+
+	timeout := defaultTimeout
+	if ymlConfig.Timeout > 0 {
+		timeout = ymlConfig.Timeout
+	}
+
+	connectTimeout := defaultConnectTimeout
+	if ymlConfig.ConnectTimeout > 0 {
+		connectTimeout = ymlConfig.ConnectTimeout
+	}
+
+	reconnectInterval := defaultReocnnectInterval
+	if ymlConfig.ReconnectInterval > 0 {
+		reconnectInterval = ymlConfig.ReconnectInterval
+	}
+
+	protoVersion := defaultProtoVersion
+	if ymlConfig.ProtoVersion > 0 {
+		protoVersion = ymlConfig.ProtoVersion
+	}
+
+	clientConfig := &gocql.ClusterConfig{
+		Hosts:             ymlConfig.Hosts,
+		Port:              ymlConfig.Port,
+		Timeout:           timeout,
+		ConnectTimeout:    connectTimeout,
+		ReconnectInterval: reconnectInterval,
+		ProtoVersion:      protoVersion,
+	}
+
+	cfg := &ClientConfig{ClusterConfig: clientConfig}
+
+	return cfg, nil
+}
+
+// CreateSessionFromClientConfigAndKeyspace Creates session from given configuration and keyspace
+func CreateSessionFromClientConfigAndKeyspace(config Config, keyspace string) (*gocql.Session, error) {
```

Comment:
please make the keyspace optional (since we want the user rather to put metadata (keyspace/schemane and table name) to dedicated method in structures. You can put that to Config structure if needed but please comment that the preference about metadata (my last pull request to cassandra SchemanName() example)